graphql/resolvers/events.js

```diff
@@ -60,54 +63,64 @@ module.exports = {
                     }
                 });
             }
-            if(!page)
-                page = 0;
-            if(!pageSize)
-                pageSize = 50;
-            if(!bikeType) {
-                bikeType = [];
-            }
-            if(!wkg) {
-                wkg = [];
+
+            page = page || 0;
+            pageSize = pageSize || 50;
+            bikeType = bikeType || [];
+            wkg = wkg || [];
+
+            // filter gender restrictions
+            const userGender = user.sex;
+            let genderFilter = {};
+
+            if (userGender === 'gender-man' || userGender === 'gender-prefer-not-to-say') {
+                genderFilter = {
+                    $and: [
+                        { privateNonBinary: false },
+                        { privateWomen: false }
+                    ]
+                };
+            } else if (userGender === 'gender-woman') {
+                genderFilter = {
+                    $or: [
+                        { privateWomen: true },
+                        { $and: [{privateWomen: false}, { privateNonBinary: false }] }
+                    ]
+                };
+            } else if (userGender === 'gender-non-binary') {
+                genderFilter = {
+                    $or: [
+                        { privateNonBinary: true },
```

Comment:
I played with the branch and I think you almost got it, but found one case that I don't think is working right. I changed my gender to non-binary but I can still see this private women only ride that Ashe made: 
![image](https://github.com/user-attachments/assets/fd235638-9aa8-47ae-8fd7-ccc2b726dd81)

I specifically went into the database and changed the values because when you host a ride you can see it regardless of what your gender is after you see it.